jaxtronomy/LensModel/Solver/lens_equation_solver.py

```diff
@@ -0,0 +1,773 @@
+                iter_num,
+                x_guess,
+                y_guess,
+                delta_init,
+                num_iter_max,
+                image_plane_vector,
+                subkey,
+            ]
+
+        init_val = [
+            x_new,
+            y_new,
+            delta_new,
+            iter_num,
+            x_guess,
+            y_guess,
+            delta_init,
+            num_iter_max,
+            image_plane_vector,
+            subkey,
+        ]
+
+        x_new, y_new, delta_new, iter_num, x_guess, y_guess, delta_init, _, _, _ = (
+            lax.while_loop(cond_fun, keep_guessing, init_val)
+        )
+
+        # Now that we've stopped iterating, we return either the new guess or the old guess
+        # If we got a good new guess, return the new guess, otherwise return the initial guess
+        def return_new(x_new, y_new, delta_new, iter_num, x_guess, y_guess, delta_init):
+            return x_new, y_new, delta_new, iter_num
+
+        def return_old(x_new, y_new, delta_new, iter_num, x_guess, y_guess, delta_init):
+            return x_guess, y_guess, delta_init, iter_num
+
+        return lax.cond(
+            delta_new < delta_init,
+            return_new,
+            return_old,
+            x_new,
+            y_new,
+            delta_new,
+            iter_num,
+            x_guess,
+            y_guess,
+            delta_init,
+        )
+
+    def image_position_analytical(
+        self,
+        x,
+        y,
+        kwargs_lens,
+        arrival_time_sort=True,
+        magnification_limit=None,
+        **kwargs_solver,
+    ):
+        """Solves the lens equation. Only supports EPL-like (plus shear) models. Uses a
+        specialized recipe that solves a one-dimensional lens equation that is easier
+        and more reliable to solve than the usual two-dimensional lens equation.
+
+        :param x: source position in units of angle, an array of positions is also supported.
+        :param y: source position in units of angle, an array of positions is also supported.
+        :param kwargs_lens: lens model parameters as keyword arguments
+        :param arrival_time_sort: bool, if True, sorts image position in arrival time (first arrival photon first listed)
+        :param magnification_limit: None or float, if set will only return image positions that have an
+         abs(magnification) larger than this number
+        :param kwargs_solver: additional kwargs to be supplied to the solver. Particularly relevant are Nmeas and Nmeas_extra
+        :returns: (exact) angular position of (multiple) images ra_pos, dec_pos in units of angle
+         Note: in contrast to the other solvers, generally the (heavily demagnified) central image will also be included, so
+         setting a proper magnification_limit is more important. To get similar behaviour, a limit of 1e-1 is acceptable
+        """
+        lens_model_list = copy.deepcopy(list(self.lensModel.lens_model_list))
```

Comment:
maybe this can be a self variable? I worry that deep copying for each function call might be an issue for JAX

Reply:
Currently this function is not jaxified since it the analytical solver uses the numba compiler. We can change this whenever we jaxify the analytical solver

Reply:
Just tested it and deepcopy does not seem to cause any issues with JAX